lib/cog/commands/bundle.ex

```diff
@@ -69,70 +75,130 @@ defmodule Cog.Commands.Bundle do
-
+  def handle_message(%{args: ["status", bundle_name]} = req, state) do
+    case Bundles.with_status_by_name(bundle_name) do
+      nil ->
+        {:error, req.reply_to, error_message({:not_found, bundle_name}), state}
+      bundle ->
+        {:reply, req.reply_to, "bundle-status", bundle, state}
+    end
+  end
+
+  def handle_message(%{args: [action, "operable"|_]} = req, state)
+      when action in ["enable", "disable"] do
+    {:error, req.reply_to, error_message({:embedded_bundle, action}), state}
+  end
+
+  def handle_message(%{args: ["enable", bundle_name|args]} = req, state) do
+    result = with {:ok, bundle} <- parse_and_find_version(bundle_name, args),
+                  :ok           <- check_for_enabled_version(bundle_name),
+                  {:ok, bundle} <- enable_bundle(bundle),
+                  do: {:ok, bundle}
+
+    case result do
+      {:ok, bundle} ->
+        {:reply, req.reply_to, "bundle-enable", Map.put(bundle, :status, :enabled), state}
+      {:error, error} ->
+        {:error, req.reply_to, error_message(error), state}
+    end
+  end
+
+  def handle_message(%{args: ["disable", bundle_name]} = req, state) do
+    result = with {:ok, bundle} <- find_enabled_bundle(bundle_name),
+                  {:ok, bundle} <- disable_bundle(bundle),
+                  do: {:ok, bundle}
+
+    case result do
+      {:ok, bundle} ->
+        {:reply, req.reply_to, "bundle-disable", Map.put(bundle, :status, :disabled), state}
+      {:error, error} ->
+        {:error, req.reply_to, error_message(error), state}
+    end
+  end
+
+  def handle_message(req, state) do
+    {:error, req.reply_to, error_message(:invalid_invocation), state}
+  end
+
+  defp enable_bundle(%BundleVersion{}=bundle_version) do
+    case Bundles.set_bundle_version_status(bundle_version, :enabled) do
+      :ok ->
+        {:ok, bundle_version}
+      error ->
+        error
+    end
+  end
+
+  defp disable_bundle(%BundleVersion{}=bundle_version) do
+    case Bundles.set_bundle_version_status(bundle_version, :disabled) do
+      :ok ->
+        {:ok, bundle_version}
+      error ->
+        error
+    end
+  end
+
+  defp parse_and_find_version(bundle_name, []) do
+    case Bundles.highest_version_by_name(bundle_name) do
+      nil ->
+        {:error, {:not_found, bundle_name}}
+      bundle ->
+        {:ok, bundle}
+    end
+  end
+
+  defp parse_and_find_version(bundle_name, [version]) do
+    case Version.parse(version) do
+      {:ok, version} ->
+        case Bundles.with_name_and_version(bundle_name, version) do
+          nil ->
+            {:error, {:not_found, bundle_name, version}}
+          bundle ->
+            {:ok, bundle}
+        end
+      :error ->
+        {:error, {:invalid_version, version}}
+    end
+  end
+
+  def check_for_enabled_version(bundle_name) do
+    case Bundles.enabled_version_by_name(bundle_name) do
+      nil ->
+        :ok
+      bundle ->
+        {:error, {:already_enabled, bundle}}
```

Comment:
You may be trying to enable a different version of a bundle that is already enabled. I could specifically handle the case where you try to enable the version of the bundle that is already enabled and return a successful response, but this seemed like a reasonable way to return a consistent response.